circup/__init__.py

```diff
@@ -1236,10 +1685,8 @@ def main(ctx, verbose, path, board_id, cpy_version):  # pragma: no cover
     # stop early if the command is boardless
     if ctx.invoked_subcommand in BOARDLESS_COMMANDS:
         return
-    if path:
-        device_path = path
-    else:
-        device_path = find_device()
+
+    device_path = get_device_path(host, password, path)
```

Comment:
I think this could be replaced by a backend function that confirms that the device is present.

Reply:
The latest commits add Backend.is_device_present() with implementations for it in both different Backends. That is now checked here: https://github.com/vladak/circup/blob/web_workflow/circup/__init__.py#L1051 

I confirmed with the latest changes that the error presented in the "missing device" case for both web workflow and disk workflow are now the same and seem to work as intended.

`get_device_path(host, password, path)` does still get called here and completes successfully even in the "missing device" case. That check and the error it produces come a little after this. Let me know if you envision that part working differently.